File: apps/mobile/lib/services/orders_service.dart
import 'api_service.dart';

class OrdersService {
  static Future<Map<String, dynamic>> create({
    required String orderNumber,
    required String staffId,
    required String branchId,
    String? shiftId,
    required String orderType,
    String? tableNumber,
    required int subtotal,
    required int taxAmount,
    required int serviceAmount,
    int discountAmount = 0,
    required int totalAmount,
    String? notes,
    required List<Map<String, dynamic>> items,
    required List<Map<String, dynamic>> paymentMethods,
  }) async {
    return ApiService.post('/orders', {
      'orderNumber': orderNumber,
      'staffId': staffId,
      'branchId': branchId,
      'shiftId': shiftId,
      'orderType': orderType,
      'tableNumber': tableNumber,
      'subtotal': subtotal,
      'taxAmount': taxAmount,
      'serviceAmount': serviceAmount,
      'discountAmount': discountAmount,
      'totalAmount': totalAmount,
      'notes': notes,
      'items': items,
      'paymentMethods': paymentMethods,
    });
  }

  static Future<List<dynamic>> getAll({String? branchId, String? status, String? shiftId}) async {
    final params = <String>[];
    if (branchId != null) params.add('branchId=$branchId');
    if (status != null) params.add('status=$status');
    if (shiftId != null) params.add('shiftId=$shiftId');
    final query = params.isNotEmpty ? '?${params.join('&')}' : '';
    return ApiService.getList('/orders$query');
  }

  static Future<List<dynamic>> getByShift(String shiftId) async {
    return ApiService.getList('/orders?shiftId=$shiftId');
  }

  static Future<Map<String, dynamic>> getById(String id) async {
    return ApiService.get('/orders/$id');
  }
}

File: apps/admin/src/services/orders.service.js
import { apiClient } from '../lib/api-client';

export const ordersService = {
    getAll: (filters = {}) => {
        const params = new URLSearchParams();
        if (filters.branchId) params.set('branchId', filters.branchId);
        if (filters.status) params.set('status', filters.status);
        if (filters.startDate) params.set('startDate', filters.startDate);
        if (filters.endDate) params.set('endDate', filters.endDate);
        const qs = params.toString();
        return apiClient.get(`/orders${qs ? `?${qs}` : ''}`);
    },
    getById: (id) => apiClient.get(`/orders/${id}`),
    getSaved: (shiftId) => apiClient.get(`/orders/saved?shiftId=${shiftId}`),
    create: (data) => apiClient.post('/orders', data),
    updateStatus: (id, status) => apiClient.put(`/orders/${id}/status`, { status }),
};
